### ISSUE TITLE
When removing languages, old URLs (with language code) aren't redirected to the new URL (without language code)

### ISSUE THREAD

**Issue by Maofree** [PrestaShop/PrestaShop#16865]
Hi
In past I've alert Prestashop to solve a SEO problem for multi-languages sites
with WPML is it possible to remove the iso_code of the default language if the site has some languages.
Normally a site starts from one languange and the others will be added after a while
so many of multi-languages sites will have the problem after the installation of the second language, because all urls of the site will change, so with the iso_code in the urls will change them.
This will add a seo problem with duplications and for external links that point to a different link.

is it possible to add an option from backoffice to decide if add or remove the iso code for the default language? all users will appreciate this feature

for example, now I need to add some languages to two sites but I cannot do it before to have found a solution of these seo errors

bye

**Screenshots**
![1](https://user-images.githubusercontent.com/653230/71170363-629c4c00-225b-11ea-8136-cdbb8b048a38.png)

**Additional information**
PrestaShop version: 1.7.6.2
PHP version: 7.x

**Comment by Maofree** [comment]
the same problem is present if I start from a site with some languages and then I remove all of them except one

**Comment by khouloudbelguith** [comment]
Hi @maofree,

PrestaShop is not compatible with PHP 7.3, in fact, PrestaShop 1.7.0 to 1.7.3 is compatible with PHP 7.1 and PrestaShop 1.7.4 & 1.7.5 & PS1.7.6 are compatible with PHP 7.2.
For more details, you can follow this link: http://build.prestashop.com/news/prestashop-1-7-is-moving-to-symfony-3-4-and-php-5-6/
Next Major release PS1.7.7 will be compatible with PHP7.3.
Please downgrade your PHP version.

I think it is same issue: https://github.com/PrestaShop/PrestaShop/issues/13737? is it?

Thanks!

**Comment by Maofree** [comment]
hi
it is not a php problem
read better the issue, it is present on all php version
try to use a site with 2-3 languages in the urls will be present the iso_code also for the default language

**Comment by Maofree** [comment]
at this moment there is only a module that remove it
https://mypresta.eu/modules/seo/remove-language-iso-code-from-urls.html
but I don't know if it can remove all posible problems from a site with complex configurations
it should be present by default in prestashop core

**Comment by khouloudbelguith** [comment]
@maofree, 

>it is not a php problem
read better the issue, it is present on all php version

It is just information about PrestaShop compatibility.

>Add an option to remove the language iso_code from the urls of the default language

Ping @PrestaShop/prestashop-product-team what do you think of this improvement?
Related to #13737. 

Thanks!

**Comment by eternoendless** [comment]
As far as I know this is not a SEO issue, only cosmetic.

1. The original page (without ISO) is redirected to the new page (make sure to set up 301 redirects in the "Traffic & SEO" page) so page rank is transferred to the new canonical.
2. Translated pages are not considered by Google as duplicate content.

**Comment by Maofree** [comment]
Hi
and the opposite case?
from multi-languages to single language
the url like
xxxx/it/rrrrr -> xxxx/rrrr
get a 404 error?

**Comment by eternoendless** [comment]
To my best understanding, once your shop becomes multilingual, all addresses without language are permanently redirected to `xxxx/it/rrrrr` even if you delete all other languages.

**Comment by Maofree** [comment]
Hi
it is not true, there is any redirect when I return to a single language from the previous urls, it will be shown a 404 error even if set the 301 redirect.

I've tested this problem on a demo site
you can do it easily:
1) use a multi-language site
2) go to the front-office site in a product page using the default language
3) disable all languages except the default one
4) now reload the product page
5) now you will get a 404 error

the problem is simple, there is not a redirect to the default url without iso code
the best solution is to remove the iso code from the url for default language, in this mode you remove all possible errors for all future Prestashop's versions, all doubts, all possible google strategy changes
in this mode no changes will be done with or without extra languages
the simplest solution is always the best.
and to remove all previous compatibility it is necessary only an option to give the possibility to add or not the iso code for the default language.

![1](https://user-images.githubusercontent.com/653230/71385178-03856100-25e6-11ea-81c1-fc66e33bc8bd.png)
![2](https://user-images.githubusercontent.com/653230/71385179-03856100-25e6-11ea-8ec4-1ab3ddc89819.png)

bye

**Comment by eternoendless** [comment]
You're right @maofree and to me that's a bug.

Short recap: when you disable other languages, old product URLs (with language code) aren't redirected to the new URL (without language code).

Ping @colinegin @marionf

**Comment by marionf** [comment]
Yes, I think I have already seen an issue like this but can't find it

**Comment by Maofree** [comment]
also Gravcms has an option do add or not the default language in the urls

![1](https://user-images.githubusercontent.com/653230/72552271-ce8fc580-3896-11ea-98a2-0ddbb78923f1.png)

**Comment by Hlavtox** [comment]
@Maofree Hi! I implemented this into Prestashop 9.0.0, you can try it in the beta. You can now choose if you want the iso prefix for default language. :-)

**Comment by Maofree** [comment]
Hi
thanks @Hlavtox 

bye
